packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -7030,12 +7090,29 @@ class Semantics extends SingleChildRenderObjectWidget {
   /// an [ExcludeSemantics] widget and then another [Semantics] widget.
   final bool excludeSemantics;
 
+  /// Whether to block user interactions for the rendering subtree.
+  ///
+  /// Setting this to true will prevent user from interacting with this
+  /// rendering object and its subtree through pointer-related
+  /// [SemanticsAction]s in assistive technologies.
+  ///
+  /// This [SemanticsNode] created from this rendering object is still focusable
+  /// by assistive technologies. Only pointer-related [SemanticsAction]s, such as
+  /// [SemanticsAction.tap] or its friends, are blocked.
+  ///
+  /// If this rendering object is merged into a parent semantics node, only the
```

Comment:
I am actually not 100% clear on how merging works here. If I have a parent SemanticsConfig with a tap handler and there are two child semantics configs: One of them has a longpress handler and the other one sets blockUserInteractions. These will all get implicitly merged into a single node, right? What actions are now allowed on that node?

Also, if I force merge (using MergeSemantics) a SemanticsNode with blockUserInteractions set into another node with an action handler, will the resulting node have its action blocked or not?

Reply:
> If I have a parent SemanticsConfig with a tap handler and there are two child semantics configs: One of them has a longpress handler and the other one sets blockUserInteractions. These will all get implicitly merged into a single node, right? What actions are now allowed on that node?

Only the action in blockUserInteractions child will be block and drop from the merging. Soe the onTap on parent and the onLongPress on the child will still be merged into the resulting node.

> if I force merge (using MergeSemantics) a SemanticsNode with blockUserInteractions set into another node with an action handler, will the resulting node have its action blocked or not?

The resulting node will not have its actions blocked, the actions in the node with blockUserInteractions will be dropped.

Reply:
Okay, both behave the same here, that's good. Thanks for the clarification!

Reply:
(I can't remember, do you have tests in place for both of these scenarios to make sure that this doesn't break?)

Reply:
yes, the test in `flutter/packages/flutter/test/widgets/semantics_test.dart`